File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/model/asset/AssetModel.java
/* Copyright (c) Koninklijke Philips N.V., 2018
 * All rights are reserved. Reproduction or dissemination
 * in whole or in part is prohibited without the prior written
 * consent of the copyright holder.
 */
package com.philips.platform.ecs.model.asset;


import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.philips.platform.ecs.prx.response.ResponseData;
import com.philips.platform.ecs.prx.response.ResponseData;

import org.json.JSONObject;

import java.io.Serializable;


/**
 * Description :
 * Project : PRX Common Component.
 * Created by naveen@philips.com on 02-Nov-15.
 */
public class AssetModel extends ResponseData implements Serializable {
    private boolean success;
    @SerializedName("data")
    @Expose
    private Data data;

    /**
     * No args constructor for use in serialization
     *
     */
    public AssetModel() {
    }

    /**
     *
     * @param data
     * @param success
     */
    public AssetModel(boolean success, Data data) {
        this.success = success;
        this.data = data;
    }

    /**
     *
     * @return
     * The success
     */
    public boolean isSuccess() {
        return success;
    }

    /**
     *
     * @param success
     * The success
     */
    public void setSuccess(boolean success) {
        this.success = success;
    }

    /**
     *
     * @return
     * The data
     */
    public Data getData() {
        return data;
    }

    /**
     *
     * @param data
     * The data
     */
    public void setData(Data data) {
        this.data = data;
    }

    @Override
    public ResponseData parseJsonResponseData(JSONObject jsonResponse) {
        if (jsonResponse != null) {
            return new Gson().fromJson(jsonResponse.toString(), AssetModel.class);

        }
        return null;




    }
}

File: Source/prx/Source/Library/PrxSample/prx/src/main/java/com/philips/cdp/prxclient/datamodels/assets/AssetModel.kt
package com.philips.cdp.prxclient.datamodels.assets

import android.os.Parcelable
import com.google.gson.Gson
import com.philips.cdp.prxclient.response.ResponseData
import kotlinx.android.parcel.Parcelize
import org.json.JSONObject

/**
 * Description :
 * Project : PRX Common Component.
 */
@Parcelize
data class AssetModel(var isSuccess: Boolean = true, var data: Data? = null) : Parcelable, ResponseData() {
    override fun parseJsonResponseData(jsonResponse: JSONObject?): ResponseData? {
        return if (jsonResponse != null) {
            Gson().fromJson(jsonResponse.toString(), AssetModel::class.java)
        } else null
    }
}